server/service/team_policies.go

```diff
@@ -183,20 +194,49 @@ func deleteTeamPoliciesEndpoint(ctx context.Context, request interface{}, svc fl
 }
 
 func (svc Service) DeleteTeamPolicies(ctx context.Context, teamID uint, ids []uint) ([]uint, error) {
+	// First check if authorized to read policies
 	if err := svc.authz.Authorize(ctx, &fleet.Policy{
 		PolicyData: fleet.PolicyData{
 			TeamID: ptr.Uint(teamID),
 		},
-	}, fleet.ActionWrite); err != nil {
+	}, fleet.ActionRead); err != nil {
 		return nil, err
 	}
 	if len(ids) == 0 {
 		return nil, nil
 	}
-	ids, err := svc.ds.DeleteTeamPolicies(ctx, teamID, ids)
+	policiesByID, err := svc.ds.PoliciesByID(ctx, ids)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "getting policies by ID")
+	}
+
+	// Then check if authorized to write policies
```

Comment:
Should you check that the returned policies in `PoliciesByID` all have `Policy.TeamID` == `teamID`.

It looks like a user could send a request with a teamID it's a member of but a set of policy ids that do not belong to such team.

Reply:
@lucasmrod do you think the auth check below covers the issue? Or do you think we should also loop over the returned policies and return an error if any `Policy.TeamID != teamID`?

Reply:
Ah I see what you are getting at now! I'll make a change